src/pages/wallet-overview/wallet-overview.spec.ts

```diff
@@ -514,5 +456,493 @@ describe("WalletOverviewPage", () => {
+      }}), "keystore");
+  });
+
+  it("should create a modal to enter the password for privatekey export", () => {
+    let alert = new MockAlert();
+    comp.currentWallet = getDummyWallet();
+
+    spyOn(alertController, "create").and.returnValue(alert);
+    spyOn(alert, "present");
+
+    comp.handleExportModalClick("DATA_TYPE", "privatekey");
+
+    expect(alertController.create).toHaveBeenCalled();
+
+    expect(alert.present).toHaveBeenCalled();
+  });
+
+  it("should not call platform since the datatype is not defined", () => {
+    spyOn(platform, "is");
+
+    comp.export("", "data", "keystore");
+
+    expect(platform.is).not.toHaveBeenCalled();
+  });
+
+  it("should use the clipboard library to copy the files when the platform is android or ios", () => {
+    spyOn(clipBoard, "copy");
+    spyOn(platform, "is").and.returnValue(true);
+
+    comp.export("clipboard", "data", "keystore");
+
+    expect(clipBoard.copy).toHaveBeenCalledWith("data");
+  });
+
+  it("should use the javascript way to copy the files to the clipboard because the platform is something else than android or ios", () => {
+    spyOn(platform, "is").and.returnValue(false);
+    spyOn(comp, "copyToClipboardWeb");
+
+    comp.export("clipboard", "data", "keystore");
+
+    expect(comp.copyToClipboardWeb).toHaveBeenCalledWith("data");
+  });
+
+  it("should show a toast at the bottom of the page", () => {
+    spyOn(comp, "showToastMessage");
+
+    comp.export("clipboard", "data", "keystore");
+
+    expect(comp.showToastMessage).toHaveBeenCalled();
+  });
+
+  it("should export the keystore as a filename without a name", () => {
+    spyOn(platform, "is").and.returnValue(false);
+    spyOn(comp, "downloadTxtFileWeb");
+
+    comp.export("file", "data", "SOMETHING_ELSE");
+
+    expect(comp.downloadTxtFileWeb).toHaveBeenCalledWith("data", "");
+  });
+
+  it("should write the file to android storage", () => {
+    comp.currentWallet = getDummyWallet();
+    spyOn(platform, "is").and.returnValue(true);
+    spyOn(comp, "writeFileMobile");
+    // Filenative throws a warning that cordova isn't loaded in this unit test. We know it isn't; just don't show it.
+    spyOn(console, "warn");
+    
+    comp.export("file", "data", "keystore");
+
+    expect(comp.writeFileMobile).toHaveBeenCalledWith(jasmine.any(String),
+                                                      jasmine.any(String),
+                                                      "data",
+                                                      {replace: true},
+                                                      "android");
+    
+  });
+
+  it("should write the file to ios storage", () => {
+    comp.currentWallet = getDummyWallet();
+    spyOn(platform, "is").and.callFake((arg) => {
```

Comment:
This is pretty much what I meant with the above comment. However the code checking the platform can be simplified to a single line:
```
return arg == "ios"
```